src/main/java/org/jabref/logic/importer/util/JsonReader.java

```diff
@@ -1,39 +1,35 @@
 package org.jabref.logic.importer.util;
 
-import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.nio.charset.StandardCharsets;
 
 import org.jabref.logic.importer.ParseException;
 
+import com.google.common.base.Charsets;
+import com.google.common.io.ByteStreams;
+import kong.unirest.json.JSONException;
 import kong.unirest.json.JSONObject;
 
 /**
  * Converts an {@link InputStream} into a {@link JSONObject}.
  */
 public class JsonReader {
 
-    public static JSONObject toJsonObject(InputStreamReader input) throws ParseException {
-        BufferedReader streamReader = new BufferedReader(input);
-        StringBuilder responseStrBuilder = new StringBuilder();
-
+    /**
+     * Converts the given input stream into a {@link JSONObject}.
+     *
+     * @return A {@link JSONObject}. An empty JSON object is returned in the case an empty stream is passed.
+     */
+    public static JSONObject toJsonObject(InputStream inputStream) throws ParseException {
         try {
-            String inputStr;
-            while ((inputStr = streamReader.readLine()) != null) {
-                responseStrBuilder.append(inputStr);
-            }
-            if (responseStrBuilder.toString().isBlank()) {
-                throw new ParseException("Empty input!");
+            String inputStr = new String(ByteStreams.toByteArray(inputStream), Charsets.UTF_8);
```

Comment:
```suggestion
            String inputStr = new String((inputStream.readAllByte()), Charsets.UTF_8);
```
This is possible since java 9